storage/range.go

```diff
@@ -504,12 +585,33 @@ func (r *Range) addReadWriteCmd(method string, args proto.Request, reply proto.R
 		log.Errorf("unable to read result for %+v from the response cache: %s", args, err)
 	}
 
+	// For read/write commands, we check the leader lease. If we know of
+	// a lease and the holder is not this replica, and we're within the
+	// lease grace period of its expiration, return NotLeaderError. In
+	// all other cases, we want to continue. Either we're the lease
+	// holder, or were the last lease holder, or we weren't the last but
+	// we'd like to become the leader because this replica is where the
+	// write pressure is happening.
+	if r.AnotherHasLeaderLease() {
+		reply.Header().SetGoError(r.newNotLeaderError())
+		return reply.Header().GoError()
+	}
+	// If we don't have the lease and nobody else does, propose the
+	// command to force an election. We want this write pressure to
+	// reorient the leader to be closer to the requesting clients.
+	if held, _ := r.HasLeaderLease(); !held {
+		// TODO(bdarnell): will proposing the command if we don't have
```

Comment:
Would be cool not to have to trigger elections manually. But that means leaving Raft forwarding in place, right?